plugins/modules/dcnm_inventory.py

```diff
@@ -516,8 +538,8 @@ def validate_input(self):
                 seed_ip=dict(required=True, type='str'),
                 role=dict(type='str',
                           choices=['leaf', 'spine', 'border', 'border_spine', 'border_gateway', 'border_gateway_spine',
-                                   'super_spine', 'border_super_spine', 'border_gateway_super_spine'],
-                          default='leaf')
+                                   'super_spine', 'border_super_spine', 'border_gateway_super_spine', 'None'],
+                          default='None')
```

Comment:
Why are we changing the default from leaf to `None`?  We still list `leaf` as the default in the documentation above.  Is this so that it's not required for `query` state?  For other states, `merged`, `overridden` how do we enforce the default `leaf` role?